storagemarket/impl/clientstates/client_fsm.go

```diff
@@ -108,6 +108,20 @@ var ClientEvents = fsm.Events{
 		}),
 	fsm.Event(storagemarket.ClientEventDealActivated).
 		From(storagemarket.StorageDealSealing).To(storagemarket.StorageDealActive),
+	fsm.Event(storagemarket.ClientEventDealSlashed).
+		From(storagemarket.StorageDealActive).To(storagemarket.StorageDealSlashed).
+		Action(func(deal *storagemarket.ClientDeal, slashEpoch abi.ChainEpoch) error {
+			deal.SlashEpoch = slashEpoch
+			return nil
+		}),
+	fsm.Event(storagemarket.ClientEventDealExpired).
+		From(storagemarket.StorageDealActive).To(storagemarket.StorageDealExpired),
```

Comment:
I don't know if it matters, or if we can deduce this from the original deal proposal, but is it useful to know when the deal expired?

Reply:
Should be determined from dealProposal.EndEpoch